flink-connector-debezium/src/main/java/com/alibaba/ververica/cdc/debezium/DebeziumSourceFunction.java

```diff
@@ -222,7 +222,15 @@ private void snapshotOffsetState() throws Exception {
 				serializedOffset = restoredOffsetState.getBytes(StandardCharsets.UTF_8);
 			}
 		} else {
-			serializedOffset = consumer.snapshotCurrentState();
+			byte[] currentState = consumer.snapshotCurrentState();
+			if (currentState == null) {
+				// the consumer has been initialized, but has not yet received any data,
+				// which means we need to return the originally restored offsets
+				serializedOffset = restoredOffsetState.getBytes(StandardCharsets.UTF_8);
```

Comment:
An NP will be thrown here when the Application started for the first time and restoredOffsetState is null .

Reply:
Hi MichaelXucf,  could you please give a test case?
In my opinion, the case you mentioned has been covered by test cases `testCheckpointAndRestore`(Step-1) both in MySQLSourceTest and PostgreSQLSourceTest. 
When the Application started for the first time and restoredOffsetState is null, debezium will mark current offset and take a snapshot. In this case, currentState won't be null.

Reply:
when try to capture a table with no data in it, it will throw the Exception.  The test case ```testCheckpointAndRestore(Step-1)``` doesn't cover this case.
You can try it with the followiing code
```java
public class MyTest {

    @Test
    public void testSource() throws Exception {
        DebeziumSourceFunction<String> sourceFunction = MySQLSource.<String>builder()
                .hostname("localhost")
                .port(3306)
                .databaseList("inventory")
                .username("root")
                .password("123456")
                .tableList(new String[]{"inventory.test"})
                .deserializer(new StringDebeziumDeserializationSchema())
                .build();

        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.enableCheckpointing(60000);
        env.getCheckpointConfig().setCheckpointingMode(CheckpointingMode.EXACTLY_ONCE);

        DataStreamSource<String> streamSource = env.addSource(sourceFunction);
        streamSource.print("mysql server");

        env.execute("mysql server cdc");
    }
}
```

create a test table in database inventory and keep it empty, then run the above test case:
```sql
CREATE TABLE `test` (
  `id` int NOT NULL,
  `name` varchar(255) DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```

<img width="1315" alt="state-001" src="https://user-images.githubusercontent.com/16697434/109969084-716f0180-7d2e-11eb-9aa8-f3373bf38688.png">
<img width="1261" alt="state-002" src="https://user-images.githubusercontent.com/16697434/109969103-77fd7900-7d2e-11eb-998c-06bc259a2e9b.png">

<img width="1704" alt="state-003" src="https://user-images.githubusercontent.com/16697434/109969181-8ea3d000-7d2e-11eb-9819-5e0b9baa69ff.png">

Reply:
MichaelXucf, you are right.
I think that is the point: capture a table that never had data.
We get offset from SourceRecord.sourceOffset() but it returns null, since there are no any records found both in snapshot phase and streaming change phase.
@wuchong would you please reopen this PR or just fix this problem in another PR?

Reply:
Please feel free to open a new pull request to fix this @MichaelXucf @dongdongking008

Reply:
Hi @wuchong  @dongdongking008, I have committed a [pull request #119](https://github.com/ververica/flink-cdc-connectors/pull/119) to fix this.